src/test/java/org/benetech/servicenet/service/impl/ImportServiceImplIntTest.java

```diff
@@ -112,153 +172,220 @@ public void shouldUpdateLocation() {
         Location newLocation = generateNewLocation();
         generateExistingLocation();
 
-        int dbSize = locationService.findAll().size();
-        Location updated = importService.createOrUpdateLocation(newLocation, EXISTING_EXTERNAL_ID, PROVIDER);
+        assertEquals(1, locationService.findAll().size());
+        importService.createOrUpdateLocation(newLocation, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize, locationService.findAll().size());
-        assertEquals(NEW_STRING, updated.getName());
+        List<LocationDTO> all = locationService.findAll();
+        assertEquals(1, all.size());
+        assertTrue(all.stream().allMatch(l -> l.getName().equals(NEW_STRING)));
     }
 
     @Test
     @Transactional
     public void shouldCreatePhysicalAddress() {
-        Location location = generateExistingLocation();
+        Location location = generateNewLocation();
         PhysicalAddress address = new PhysicalAddress().address1(NEW_STRING).city(NEW_STRING).stateProvince(NEW_STRING)
-            .location(location);
-        em.flush();
-        em.refresh(location);
+            .attention(NEW_STRING).country(NEW_STRING).postalCode(NEW_STRING).region(NEW_STRING);
+        location.setPhysicalAddress(address);
 
-        int dbSize = physicalAddressService.findAll().size();
-        importService.createOrUpdatePhysicalAddress(address, location);
+        assertEquals(0, physicalAddressService.findAll().size());
+        importService.createOrUpdateLocation(location, EXISTING_EXTERNAL_ID, PROVIDER);
 
-        assertEquals(dbSize + 1, physicalAddressService.findAll().size());
+        List<PhysicalAddressDTO> all = physicalAddressService.findAll();
+        assertEquals(1, all.size());
+        assertTrue(all.stream().allMatch(pa -> pa.getLocationId() != null 
```

Comment:
I think it's better to extract the var and assert all individually, to point out specifically what goes wrong if the test won't pass in the future.